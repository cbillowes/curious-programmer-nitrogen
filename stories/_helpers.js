import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page