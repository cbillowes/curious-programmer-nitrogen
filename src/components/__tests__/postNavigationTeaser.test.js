import React from 'react'
import PostNavigationTeaser from '../postNavigationTeaser'
import { getSnapshot } from './_helpers'

describe(`Post navigation teaser`, () => {
  it(`should not render without previous post`, () => {
    const teaser = (
      <PostNavigationTeaser
        previous={{}}
      />
    )
    const tree = getSnapshot(teaser)
    expect(tree).toMatchSnapshot()
  })

  it(`should not render without a next post`, () => {
    const teaser = (
      <PostNavigationTeaser
        next={{}}
      />
    )
    const tree = getSnapshot(teaser)
    expect(tree).toMatchSnapshot()
  })

  it(`should render`, () => {
    const previousDate = new Date(`2018-04-01T00:00:00Z`)
    const nextDate = new Date(`2019-04-01T00:00:00Z`)
    const previous = {
      slug: `/previous`,
      title: `Don't panic.`,
      excerpt: `Earthman Arthur Dent is rescued by his friend, Ford Prefect—an alien researcher for the titular Hitchhiker's Guide to the Galaxy, an enormous work providing information about every planet in the universe—from the Earth just before it is destroyed by the alien Vogons.`,
      limitExcerpt: 10,
      date: { previousDate },
      author: `Pixie`,
      ttr: 1,
      tags: `douglas,adams,novel`,
    }
    const next = {
      slug: `/next`,
      title: `Come to the dark side.`,
      excerpt: `Luke Skywalker's peaceful and solitary existence gets upended when he encounters Rey, a young woman who shows strong signs of the Force. Her desire to learn the ways of the Jedi forces Luke to make a decision that changes their lives forever. Meanwhile, Kylo Ren and General Hux lead the First Order in an all-out assault against Leia and the Resistance for supremacy of the galaxy.`,
      limitExcerpt: 10,
      date: { nextDate },
      author: `Elby`,
      ttr: 2,
      tags: `star,wars`,
    }
    const teaser = (
      <PostNavigationTeaser
        previous={previous}
        next={next}
      />
    )
    const tree = getSnapshot(teaser)
    expect(tree).toMatchSnapshot()
  })
})
