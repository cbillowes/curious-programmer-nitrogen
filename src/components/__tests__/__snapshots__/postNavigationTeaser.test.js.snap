// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post navigation teaser should render all that is required 1`] = `
<nav
  className="post navigation teaser"
>
  <aside
    className="previous"
  >
     
    <a
      className="nav-link"
      href="/previous"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Previous"
    >
      <h1>
        Previous
      </h1>
      <div
        className="body"
      >
        <span
          className="text"
        >
          Excerpt
        </span>
      </div>
      <div
        className="metadata"
      >
        posted on
         
        <time
          dateTime={"2018-04-01T00:00:00.000Z"}
        >
          Sunday, 1 April 2018
        </time>
         
        <span
          data-container="metadata-author"
        >
          by Clarice Bouwer
        </span>
         
        <span
          data-container="metadata-ttr"
        >
          (Est. 12 minute read)
        </span>
      </div>
    </a>
     
    <div
      className="tags"
    >
       
      <a
        className="tag"
        href="/tag/douglas"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Douglas"
      >
        douglas
      </a>
       
       
      <a
        className="tag"
        href="/tag/adams"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Adams"
      >
        adams
      </a>
       
    </div>
  </aside>
  <aside
    className="next"
  >
     
    <a
      className="nav-link"
      href="/next"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Next"
    >
      <h1>
        Next
      </h1>
      <div
        className="body"
      >
        <span
          className="text"
        >
          Excerpt
        </span>
      </div>
      <div
        className="metadata"
      >
        posted on
         
        <time
          dateTime={"2018-04-01T00:00:00.000Z"}
        >
          Sunday, 1 April 2018
        </time>
         
        <span
          data-container="metadata-author"
        >
          by Clarice Bouwer
        </span>
         
        <span
          data-container="metadata-ttr"
        >
          (Est. 15 minute read)
        </span>
      </div>
    </a>
     
    <div
      className="tags"
    >
       
      <a
        className="tag"
        href="/tag/douglas"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Douglas"
      >
        douglas
      </a>
       
       
      <a
        className="tag"
        href="/tag/adams"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Adams"
      >
        adams
      </a>
       
    </div>
  </aside>
</nav>
`;

exports[`Post navigation teaser should render everything else 1`] = `
<nav
  className="post navigation teaser"
>
  <aside
    className="previous"
  >
     
    <a
      className="nav-link"
      href="/previous"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Don't panic."
    >
      <h1>
        Don't panic.
      </h1>
      <div
        className="body"
      >
        <span
          className="text"
        >
          Earthman Arthur Dent is rescued by his friend, Ford Prefectâ€”an...
        </span>
      </div>
      <div
        className="metadata"
      >
        posted on
         
        <time
          dateTime={"2018-04-01T00:00:00.000Z"}
        >
          Sunday, 1 April 2018
        </time>
         
        <span
          data-container="metadata-author"
        >
          by Pixie
        </span>
         
        <span
          data-container="metadata-ttr"
        >
          (Est. 1 minute read)
        </span>
      </div>
    </a>
     
    <div
      className="tags"
    >
       
      <a
        className="tag"
        href="/tag/douglas"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Douglas"
      >
        douglas
      </a>
       
       
      <a
        className="tag"
        href="/tag/adams"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="Adams"
      >
        adams
      </a>
       
    </div>
  </aside>
  <aside
    className="next"
  >
     
    <a
      className="nav-link"
      href="/next"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Come to the dark side."
    >
      <h1>
        Come to the dark side.
      </h1>
      <div
        className="body"
      >
        <span
          className="text"
        >
          Luke Skywalker's peaceful and solitary existence gets upended when he...
        </span>
      </div>
      <div
        className="metadata"
      >
        posted on
         
        <time
          dateTime={"2019-04-01T00:00:00.000Z"}
        >
          Monday, 1 April 2019
        </time>
         
        <span
          data-container="metadata-author"
        >
          by Elby
        </span>
         
        <span
          data-container="metadata-ttr"
        >
          (Est. 2 minute read)
        </span>
      </div>
    </a>
     
    <div
      className="tags"
    >
       
      <a
        className="tag"
        href="/tag/star"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="star"
      >
        star
      </a>
       
       
      <a
        className="tag"
        href="/tag/wars"
        onClick={[Function]}
        onMouseEnter={[Function]}
        title="wars"
      >
        wars
      </a>
       
    </div>
  </aside>
</nav>
`;
